src/algebra/polynomial/big_operators.lean

```diff
@@ -148,28 +204,41 @@ lemma nat_degree_multiset_prod [nontrivial R] (s : multiset (polynomial R))
   (h : (0 : polynomial R) ∉ s) :
   nat_degree s.prod = (s.map nat_degree).sum :=
 begin
-  revert h,
-  refine s.induction_on _ _,
-  { simp },
-  intros p s ih h,
-  rw [multiset.mem_cons, not_or_distrib] at h,
-  have hprod : s.prod ≠ 0 := multiset.prod_ne_zero h.2,
-  rw [multiset.prod_cons, nat_degree_mul (ne.symm h.1) hprod, ih h.2,
-    multiset.map_cons, multiset.sum_cons]
+  rw [← s.map_id, nat_degree_multiset_prod'], simp,
```

Comment:
non-terminal `simp` here.

Reply:
This proof becomes much simpler if you remove the `.map f` from the other lemmas.